clickhouse/base/sslsocket.h

```diff
@@ -53,9 +54,21 @@ class SSLSocket : public Socket {
     std::unique_ptr<OutputStream> makeOutputStream() const override;
 
 private:
+    std::unique_ptr<SSLContext> context_;
     std::unique_ptr<SSL, void (*)(SSL *s)> ssl_;
 };
 
+class SSLSocketFactory : public NonSecureSocketFactory {
+public:
+    ~SSLSocketFactory() override;
+
+#if defined(WITH_OPENSSL)
```

Comment:
The whole file is not event visible to the build-system, if `WITH_OPENSSL` is not defined. Please remove the `#if defined/#endif` braces.